src/librustdoc/clean/render_macro_matchers.rs

```diff
@@ -63,7 +63,21 @@ fn snippet_equal_to_token(tcx: TyCtxt<'_>, matcher: &TokenTree) -> Option<String
 
     // Create a Parser.
     let psess = ParseSess::new(rustc_driver::DEFAULT_LOCALE_RESOURCES.to_vec());
-    let file_name = source_map.span_to_filename(span);
+
+    // Get the original filename and ensure we're using a fresh, unmapped filename
+    // to avoid trying to remap an already remapped filename
+    let file_name = match source_map.span_to_filename(span) {
```

Comment:
I wonder if we even need the filename here, normally it would be used when displaying warning, errors, debuginfo, but none of this is done here, every errors is even cancel below.

We could just use `FileName::macro_expansion_source_code(&snippet)` and call it a day.

Reply:
We can & we should, but we need the filename variable for `rustc_parse::new_parser_from_source_str`, no?

Reply:
Yes, and I think something like this should do it. The rest can just be removed.
```suggestion
    let file_name = FileName::macro_expansion_source_code(&snippet);
```

Reply:
Ack. and done!